geometry/src/vector.rs

```diff
@@ -105,6 +105,19 @@ impl Vector {
     pub fn magnitude(self) -> f64 {
         (self.x.powi(2) + self.y.powi(2)).sqrt()
     }
+
+    /// Returns unit vector of this vector, i.e. a vector with the same direction and a magnitude of 1
+    pub fn unit(self) -> Vector {
+        let magnitude = self.magnitude();
+        assert!(
+            magnitude != 0.0,
+            "Attempted to take the unit vector of a zero vector (0, 0), which is undefined"
+        );
+        Vector {
```

Comment:
Why not make use of our handy `Div` impl: `self / magnitude`

Reply:
Addressed in https://github.com/myelin-ai/engine/pull/55/commits/0aa3e239d47b8f2a95f77b445c2121eec714b3d9